contracts/Smt.metasol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+library SMT{{hashname}} {
+    bytes32 constant internal EMPTY_SUBTREE = 0x00;
+
+    struct MerkleTree {
+        uint8 depth;
+        mapping(uint256 => bytes32) emptySubtreeCache;
+        // tree level => index inside level => element hash
+        mapping(uint256 => mapping(uint256 => bytes32)) nodes;
+        mapping(uint256 => bytes) elementData;
+    }
+
+    using SMT{{hashname}} for MerkleTree;
+
+    function initialize(MerkleTree storage tree, uint8 _depth) internal {
+        require(tree.depth == 0, "Cannot initialize twice");
+        require(_depth > 0, "Depth must be non-zero");
+        tree.emptySubtreeCache[0] = {{hash}}("");
+        // current depth == 0
+        tree._calculateEmptySubtreeHash(_depth);
+        tree.depth = _depth;
+    }
+
+    function totalElements(MerkleTree storage tree) internal view returns (uint256) {
+        // 1<<depth == 2**depth
+        // max(totalElements) == 2**255
+        return 1<<tree.depth;
+    }
+
+    function increaseDepth(MerkleTree storage tree, uint8 depthDifference) internal {
+        require(tree.depth > 0, "Tree must be initialized");
+        require(depthDifference > 0, "depthDifference must be non-zero");
+        uint8 oldDepth = tree.depth;
+        // overflow protection done by solidity 0.8
+        uint8 newDepth = oldDepth + depthDifference;
+
+        tree._calculateEmptySubtreeHash(newDepth);
+
+        for (uint8 level = oldDepth+1; level <= newDepth; level = inc(level)) {
```

Comment:
suppose:
oldDepth = 254
depthDifference = 1
then newDepth = 255
inside the for loop, level starts from 255, then increased by 1 in the function "inc(uint8)". Due to wrapping mode, overflow is not catchable, so level starts again from 0 counting to 255, again at 255 overflow happens, and makes it zero again. Thus, the for loop will be counting indefinitely.